server/controllers/orderController.js

```diff
@@ -3,14 +3,22 @@ const AccessLog = require('../models/AccessLog');
 
 exports.createOrder = async (req, res) => {
   try {
-    // Create a new order object based on request body
-    const newOrder = new Order(req.body);
+    const { products, quantities, shipmentId, paymentId } = req.body;
+
+    // Create a new order object
+    const newOrder = new Order({
+      products,
+      quantities,
+      shipmentId,
+      paymentId,
+      userId: req.user.id, // Assuming userId is available in the request
```

Comment:
True if you are calling this function from another internal API that has the userId check